src/Panel/PageCreateDialog.php

```diff
@@ -194,6 +190,21 @@ public function load(): array
 		];
 	}
 
+	/**
+	 * Temporary model for the page to
+	 * be created, used to properly render
+	 * the blueprint for fields
+	 */
+	public function model(): Page
+	{
+		return $this->model ??= Page::factory([
+			'slug'     => 'new',
+			'template' => $this->template,
+			'model'    => $this->template,
+			'parent'   => $this->parent
```

Comment:
@bastianallgeier I guess this needs to be

```suggestion
			'parent'   => $this->parent instanceof Page ? $this->parent : null
```

Since `Page::parentModel()` will automatically fall back to Site if no parent is set.